idaes/generic_models/unit_models/distillation/tray_column.py

```diff
@@ -285,35 +376,157 @@ def initialize(self, state_args_feed=None, state_args_liq=None,
                              "being used for initialization.")
             solver = get_default_solver()
 
-        feed_flags = self.tray[self.config.feed_tray_location].initialize()
+        feed_flags = self.feed_tray.initialize()
 
         self.propagate_stream_state(
-            source=self.tray[self.config.feed_tray_location].vap_out,
+            source=self.feed_tray.vap_out,
             destination=self.condenser.inlet)
 
         self.condenser.initialize()
 
         self.propagate_stream_state(
-            source=self.tray[self.config.feed_tray_location].liq_out,
+            source=self.feed_tray.liq_out,
             destination=self.reboiler.inlet)
 
         self.reboiler.initialize()
 
-        for i in self.tray_index:
+        # initialize the rectification section
+        for i in self._rectification_index:
             self.propagate_stream_state(
                 source=self.condenser.reflux,
-                destination=self.tray[i].liq_in)
+                destination=self.rectification_section[i].liq_in)
+            self.propagate_stream_state(
+                source=self.feed_tray.vap_out,
+                destination=self.rectification_section[i].vap_in)
+            if i == 1:
+                rect_liq_flags = self.rectification_section[i]. \
+                    initialize(hold_state_liq=True, hold_state_vap=False)
+            elif i == len(self._rectification_index):
+                rect_vap_flags = \
+                    self.rectification_section[i]. \
+                    initialize(hold_state_liq=False, hold_state_vap=True)
+            else:
+                self.rectification_section[i].initialize()
+
+        # initialize the stripping section
+        for i in self._stripping_index:
+            self.propagate_stream_state(
+                source=self.feed_tray.liq_out,
+                destination=self.stripping_section[i].liq_in)
             self.propagate_stream_state(
                 source=self.reboiler.vapor_reboil,
-                destination=self.tray[i].vap_in)
-            self.tray[i].initialize()
+                destination=self.stripping_section[i].vap_in)
+            if i == self.config.feed_tray_location + 1:
+                strip_liq_flags = self.stripping_section[i]. \
+                    initialize(hold_state_liq=True, hold_state_vap=False)
+            elif i == self.config.number_of_trays:
+                strip_vap_flags = self.stripping_section[i]. \
+                    initialize(hold_state_liq=False, hold_state_vap=True)
+            else:
+                self.stripping_section[i].initialize()
+
+        # Create temp block to add rectification section and expanded arcs
+        self._temp_block = Block()
```

Comment:
It might be worth adding a couple of comments here to explain the process for someone reading the code. I would suggest explaining the need for the temporary block, and quick overview of the routine (add stripping section and solve, add rectification and solve, etc.). This is mainly just so that a reader knows what is happening before reading the code, rather than having to decipher the code to work out what is happening.

Reply:
Along this lines of @andrewlee94's comment, it looks like you are solving the rectification section, then the rectification section *with* the stripping section (even though the two are decoupled).  What would you think of the following:
```python
self._temp_block = Block()
self._temp_block.rectification = Block()
# add the rectification section here)
solver.solve(self._temp_block.rectification)

self._temp_block.stripping = Block()
# add the stripping section here)
solver.solve(self._temp_block.stripping)

# add the feed tray to _temp_block
solver.solve(self._temp_block)
```

Reply:
I agree @jsiirola. I was just being lazy on my first try. I will implement this in the next commit.